### PR TITLE
Make RAMJobStore.RemoveJobInternal return true even if job has no triggers

### LINKED ISSUE #1579: DeleteJob returns false when RAMJobStore is used
**Describe the bug**

According to its summary, `IScheduler.DeleteJob` returns `true if the Job was found and deleted.`
However, when I use RAMJobStore and the job has no triggers, it returns false. The job does indeed get deleted, it's just the return value that is wrong.

**Version used**

3.4.0

**To Reproduce**

```c#
var group = "group";
var name = "name";

var job = JobBuilder.Create()
    .StoreDurably()
    .OfType(typeof(TestJob))
    .WithIdentity(name, group)
    .RequestRecovery()
    .Build();

await m_scheduler.AddJob(job, false);
var jobKey = new JobKey(name, group);
    
var get1 = await m_scheduler.GetJobDetail(jobKey);  //returns the job, as expected

var succ = await m_scheduler.DeleteJob(jobKey);    //returns false for some reason
        
var get2 = await m_scheduler.GetJobDetail(jobKey);      //returns null as expected
```

where m_scheduler is obtained like so:
```c#
var factory = new StdSchedulerFactory ();
IScheduler m_scheduler = await m_factory.GetScheduler();
```
with no config files (quartz uses the default config)

**Expected behavior**

`IScheduler.DeleteJob` returns `true`, as it does with AdoJobStore.

**Additional context**
I believe the problem is in RAMJobStore.cs:

```c#
private bool RemoveJobInternal(JobKey jobKey)
{
    lock (lockObject)
    {
        bool found = false;
        var triggersForJob = GetTriggersForJobInternal(jobKey);
        foreach (IOperableTrigger trigger in triggersForJob)
        {
            RemoveTriggerInternal(trigger.Key);
            found = true;
        }

        if (jobsByKey.Remove(jobKey) || found)
        {
            if (jobsByGroup.TryGetValue(jobKey.Group, out var grpMap))
            {
                if (grpMap.Remove(jobKey) && grpMap.Count == 0)
                {
                    jobsByGroup.Remove(jobKey.Group);
                }
            }
        }
        return found;
    }
}
```
`found` is what eventually gets returned from `IScheduler.DeleteJob`, but it only gets set to `true` if the job has any triggers.

### PR DESCRIPTION
**Pull request by Sobotkama**
Fixes #1579

### CODE REVIEW

File: src/Quartz.Tests.Unit/SchedulerTest.cs [hunk 1]
```diff
@@ -325,5 +325,32 @@ public async Task ReschedulingTriggerShouldKeepOriginalNextFireTime()
 
             await scheduler.Shutdown(true);
         }
+        
+        [Test]
+        public async Task TestJobDeleteReturnValue()
```
[the remark this answers is not in the document]
Comment (lahma) [reply src/Quartz.Tests.Unit/SchedulerTest.cs:330]: I think @drieseng has a valid point. @Sobotkama can you please move this to `RamJobStoreTest`? I think the shorted version would be something like:

```c#
[Test]
public async Task TestJobDeleteReturnValue()
{
    var job = JobBuilder.Create<TestJob>()
        .WithIdentity("j5")
        .StoreDurably()
        .Build();
    
    var store = new RAMJobStore();
    await store.StoreJob(job, false);

    var deleteSuccess = await store.RemoveJob(job.Key);
    Assert.IsTrue(deleteSuccess, "Expected DeleteJob to return True when deleting an existing job");

    deleteSuccess = await store.RemoveJob(job.Key);
    Assert.IsFalse(deleteSuccess, "Expected DeleteJob to return False when deleting an non-existing job");
}
```

[the remark this answers is not in the document]
Comment (Sobotkama) [reply src/Quartz.Tests.Unit/SchedulerTest.cs:330]: Yeah, good point, I'll do that as soon as I get to a computer (tomorrow)

[the remark this answers is not in the document]
Comment (Sobotkama) [reply src/Quartz.Tests.Unit/SchedulerTest.cs]: Done

**Comment by lahma** [comment]
Thank you!

**Pull request merged**